content/docs/esc/other-integrations/cloudflare.md

```diff
@@ -0,0 +1,254 @@
+**New to Pulumi ESC?** Complete the [Getting Started tutorial](https://www.pulumi.com/docs/esc/get-started/)
+{{< /notes >}}
+
+### 1. Create an ESC Environment
+
+Use the Pulumi ESC CLI to create and configure an Environment. Alternatively, to use the Pulumi Cloud console follow the [console instructions](https://www.pulumi.com/docs/esc/get-started/create-environment/#create-via-the-console).
+
+```bash
+esc login # if needed
+
+## create a new ESC Environment
+ESC_ENV=my-dev-environment
+esc env init ${ESC_ENV}
+esc env ls | grep ${ESC_ENV}
+```
+
+Edit the Environment in your terminal:
+
+```bash
+esc env edit ${ESC_ENV}
+```
+
+Paste the contents below in the editor and replace the values. Note: the API token is declared as a secret. Once the Environment is saved, Pulumi will encrypt its value and replace it with ciphertext.
+
+```yaml
+values:
+  environmentVariables:
+    CLOUDFLARE_API_TOKEN:
+      fn::secret: abc123abc123abc123abc123abc123
+    CLOUDFLARE_ACCOUNT_ID: abc123abc123
+```
+
+Now that the Pulumi ESC Environment is created, it can be consumed in a variety of ways, such as running other shell commands without having to set the environment variables locally first.
+
+The `esc run` command opens the Environment you previously created, sets the specified environment variables, and then uses those environment variables in the context of the `wrangler` commands.
+
+### 2. Use ESC with `wrangler whoami`
+
+Log into your Cloudflare account without needing to manage the credentials directly in your shell:
+
+```bash
+# ensure you're currently not logged in
+npx wrangler logout
+
+# retrieve esc environment and authenticate programmatically
+esc run ${ESC_ENV} npx wrangler whoami
+```
+
+Because we are running `wrangler` in non-interactive mode, it requires a Cloudflare API token and account ID for authentication. The `wrangler whoami` retrieves details about the provided credentials.
+
+For additional options and details, see `esc run --help`.
+
+### 3. Use ESC to create your `.dev.vars` file
+
+The `.dev.vars` file is located in the root of your wrangler project to define secrets used when running `wrangler dev`.
+
+{{< notes type="info" >}}
+**Recommendation** Use a dedicated ESC Environment for your `.dev.vars` definitions because the esc `dotnev` format flag does not support property paths.
+{{< /notes >}}
+
+Create a new ESC Environment:
+
+```bash
+ESC_ENV=my-dev-vars-environment
+esc env init ${ESC_ENV}
+```
+
+Edit the environment in your terminal:
+
+```bash
+esc env edit ${ESC_ENV}
+```
+
+Paste the contents below in the editor:
+
+```yaml
+values:
+  environmentVariables:
+    TOP_SECRET:
+      fn::secret: "the moon is made of cheese"
+```
+
+Per [Cloudflare documentation](https://developers.cloudflare.com/workers/configuration/secrets/#secrets-in-development), the `.dev.vars` file should be formatted like a `dotenv` file.
+
+```bash
+# generate the `.dev.vars` file:
+esc env open ${ESC_ENV} --format dotenv > .dev.vars
```

Comment:
Is it possible to do it via the files property in esc? That could be additional option that we could list here - it showcases what folks can do with files

Reply:
Yes, I'll update the example.

Reply:
I added this, please take a look.